src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Solution/EntrypointResolver.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.FileSystemGlobbing;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Steps.Solution
+{
+    public class EntrypointResolver : IEntrypointResolver
+    {
+        public IEnumerable<IProject> GetEntrypoints(IEnumerable<IProject> projects, IReadOnlyCollection<string> names)
+        {
+            if (projects is null)
+            {
+                throw new ArgumentNullException(nameof(projects));
+            }
+
+            if (names is null)
+            {
+                throw new ArgumentNullException(nameof(names));
+            }
+
+            var matcher = new Matcher(StringComparison.OrdinalIgnoreCase);
+
+            matcher.AddIncludePatterns(names);
+
+            var count = 0;
+            var firstProject = default(IProject);
+
+            foreach (var project in projects)
+            {
+                firstProject = count == 0 ? project : null;
+
+                if (matcher.Match(project.FileInfo.Name).HasMatches)
+                {
+                    count++;
+                    yield return project;
+                }
+            }
+
+            if (names.Count == 0 && firstProject is not null)
```

Comment:
If `names.Count == 0`, wouldn't that mean nothing will match? If so, `firstProject` will just be the last element of `projects` (since `count` will never increment). It seems like just yield returning `projects.LastOrDefault()` when `name.Count` is 0 would be a more readable way of getting that behavior. 

I'm not sure that's the behavior we want, though. If nothing matches, why arbitrarily pick the last project of the solution as the entry point? Shouldn't we return an empty enumerable and prompt the user to select an entry point in that case similar to the behavior we have today?

Reply:
It was picking this first project. I cannot remember why I did that.... I remember at the time it made sense, but I've removed it as it doesn't seem to make sense.